File: batches_c/batch_06_loop_type_change/compare.c
#include <stdio.h>

int sum_list(int *nums, int len) {
    int total = 0;
    for (int i = 0; i < len; i++) { total += nums[i]; }
    return total;
}

int main() {
    int nums[] = {1, 2, 3, 4, 5};
    printf("%d\n", sum_list(nums, 5));
    return 0;
}

File: batches_csharp/batch_06_loop_type_change/compare.cs
using System;

class SumList {
    static int Sum(int[] nums) {
        int total = 0, i = 0;
        while (i < nums.Length) { total += nums[i]; i++; }
        return total;
    }

    static void Main() {
        int[] nums = {1, 2, 3, 4, 5};
        Console.WriteLine(Sum(nums));
    }
}
